ui/shared/ad/CustomAdBanner.tsx

```diff
@@ -0,0 +1,61 @@
+import { Flex, chakra, Tooltip, Image } from '@chakra-ui/react';
+import { useQuery } from '@tanstack/react-query';
+import React, { useState, useEffect } from 'react';
+
+import type { AdCustomConfig } from 'types/client/ad';
+
+import config from 'configs/app';
+import type { ResourceError } from 'lib/api/resources';
+import useFetch from 'lib/hooks/useFetch';
+
+import useIsMobile from '../../../lib/hooks/useIsMobile';
+
+const CustomAdBanner = ({ className }: { className?: string }) => {
+  const isMobile = useIsMobile();
+
+  const feature = config.features.adsBanner;
+  const configUrl = (feature.isEnabled && feature.provider === 'custom') ? feature.configUrl : '';
+
+  const apiFetch = useFetch();
+  const { data: adConfig } = useQuery<unknown, ResourceError<unknown>, AdCustomConfig>(
+    [ 'custom-configs' ],
+    async() => apiFetch(configUrl),
+    {
+      enabled: feature.isEnabled && feature.provider === 'custom',
+      staleTime: Infinity,
+    });
+  const interval = adConfig?.interval || 60000;
+  const banners = adConfig?.banners || [];
+
+  const [ currentBannerIndex, setCurrentBannerIndex ] = useState(0);
+  useEffect(() => {
+    if (banners.length === 0) {
+      return;
+    }
+    const timer = setInterval(() => {
+      setCurrentBannerIndex((prevIndex) => (prevIndex + 1) % banners.length);
+    }, interval);
+
+    return () => {
+      clearInterval(timer);
+    };
+  }, [ interval, banners.length ]);
+
+  if (banners.length === 0) {
```

Comment:
this part causes a layout shift, because before you fetch config it shows nothing. maybe it's better to require an array of banners to be not empty (for the empty case we have NEXT_PUBLIC_AD_BANNER_PROVIDER=none) and reserve a place for the banner

Reply:
Perhaps it's just right not to occupy a position when the array is empty? The effect is equivalent to NEXT_PUBLIC_AD_BANNER_PROVIDER=none

Reply:
What I tried to say, empty array is equal to no ad, so there shouldn’t be an empty config for a custom banner. In that case you can remove this condition, and reserve ad place while fetching a config. Otherwise (like in your demo) you can see, that page renders without an ad, then ad appears and it causes a layout shift (latest blocks and txs goes down), which is bad for UX

Reply:
Well, I will reserve the height of the advertisement before obtaining the custom configuration file.